data_node/upload_manager.go

```diff
@@ -0,0 +1,123 @@
+package datanode
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"path"
+	"strconv"
+
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/errors"
+	"github.com/julienschmidt/httprouter"
+)
+
+// NewUploadManager A function to create a new upload manager
+func NewUploadManager() UploadManager {
+	return UploadManager{
+		fileBase:  make(map[string]FileInfo),
+		logPrefix: "[Upload-Manager]",
+	}
+}
+
+// HandleUpload is upload endpoint handler
+func (um *UploadManager) HandleUpload(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	log.Println(um.logPrefix, fmt.Sprintf("Received request from %s", r.RemoteAddr))
+	reqType := r.Header.Get("Request-Type")
+
+	switch reqType {
+	case "INIT":
+		um.handleInitialUpload(w, r)
+	case "APPEND":
+
+	default:
+		log.Println(um.logPrefix, r.RemoteAddr, fmt.Sprintf("request-type header value undefined - %s", reqType))
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Request-Type header value is not undefined"))
+	}
+}
+
+func (um *UploadManager) handleInitialUpload(w http.ResponseWriter, r *http.Request) {
+	log.Println(um.logPrefix, r.RemoteAddr, "Received INIT request")
+
+	err := um.checkInitialUploadHeaders(&r.Header, false)
+
+	if err != nil {
+		log.Println(um.logPrefix, r.RemoteAddr, err)
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
+		return
+	}
+
+	filesize, _ := strconv.ParseInt(r.Header.Get("Content-Length"), 10, 64)
```

Comment:
We shouldn't ignore this error.

Reply:
I ignored it safely because I already checked it in validate headers function